include/deal.II/fe/fe_update_flags.h

```diff
@@ -410,6 +438,37 @@ namespace internal
       std::vector<DerivativeForm<1,spacedim,dim> > inverse_jacobians;
 
       /**
+       * Array of the derivatives of the Jacobian matrices at the
+       * quadrature points, pushed forward to the real cell coordinates.
+       */
+      std::vector<Tensor<3,spacedim> > jacobian_pushed_forward_grads;
+
+      /**
+       * Array of the second derivatives of the Jacobian matrices at the
+       * quadrature points.
+       */
+      std::vector<DerivativeForm<3,dim,spacedim> > jacobian_hessians;
+
+      /**
+       * Array of the  second derivatives of the Jacobian matrices at the
+       * quadrature points, pushed forward to the real cell coordinates.
+       */
+      std::vector<Tensor<4,spacedim> > jacobian_pushed_forward_hessians;
```

Comment:
The best description is in the comment above FEValues::jacobian_pushed_forward_*(). It's something like $G_{ijkl}=\frac{d^2J_{iJ}}{d\hat x_K d\hat x_L} (J_{jJ})^{-1} (J_{kK})^{-1}(J_{lL})^{-1}$.

The use I had in mind was in FE::fill_fe_*_values(), to account for terms that depend on the jacobian derivatives of various orders. These are always needed in the "pushed_forward" form. 

I don't think the pushed_forward versions are much needed outside the library. They are to be computed by the mapping and used by the FE, mostly independent of the subclass combinations. Is there a way to implement them in the superclass of Mapping::InternalData so that:
1) they are updated only when the update_flags for the shape_nth_derivative that needs them is set and
2) they are always passed to FE::fill_fe_*_values?